Makefile

```diff
@@ -107,10 +108,20 @@ images: $(foreach dir,$(IMAGE_DIRS),image-$(dir))
 # Rules for building and installing binaries, or building docker images, and cleaning up.
 #
 
+KERNEL_INCLUDE_DIRS = /include \
+                      /include/uapi \
+                      /include/generated/uapi \
+                      /arch/x86/include \
+                      /arch/x86/include/uapi \
+                      /arch/x86/include/generated/uapi
+
+KERNEL_INCLUDES := $(foreach kernel_dir,$(KERNEL_SRC_DIR) $(KERNEL_BUILD_DIR),$(addprefix -I,$(wildcard $(addprefix $(kernel_dir),$(KERNEL_INCLUDE_DIRS)))))
+
 libexec/%.o: elf/%.c
+	@if [ -z $(KERNEL_INCLUDES) ]; then echo "Cannot build $@: invalid KERNEL_SRC_DIR=$(KERNEL_SRC_DIR)"; exit 1; fi
 	$(Q)echo "Building $@"
 	$(Q)mkdir -p libexec
-	$(CLANG) -I$(KERNEL_SRC_DIR)/arch/x86/include -I$(KERNEL_SRC_DIR)/arch/x86/include/uapi -I$(KERNEL_SRC_DIR)/include -O2 -Wall -target bpf -c $< -o $@
+	$(CLANG) -nostdinc -D __KERNEL__ $(KERNEL_INCLUDES) -O2 -Wall -target bpf -c $< -o $@
```

Comment:
Only one nitpick: since you're touching this line, can you also properly prefix it with $(Q), so it's consistent with the rest of the Makefile wrt. suppressed echoing vs. Q not being overridden ?

Otherwise this is good.

Reply:
$(Q) added